src/web/components/Search/SearchBar.tsx

```diff
@@ -0,0 +1,37 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import clsx from 'clsx';
+
+import './SearchBar.scss';
+
+type SearchBarContainerProps = React.PropsWithChildren<{ className?: string }>;
+export function SearchBarContainer({ children, className }: SearchBarContainerProps) {
+  return <div className={clsx('search-bar', className)}>{children}</div>;
+}
+
+type SearchBarInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
+  inputClassName?: string;
+  fullBorder?: boolean;
+};
+export function SearchBarInput({
+  className,
+  inputClassName,
+  fullBorder,
+  ...rest
+}: SearchBarInputProps) {
+  return (
+    <div className={clsx('search-bar-input-container', className, { 'full-border': fullBorder })}>
+      <FontAwesomeIcon icon='search' className='search-icon' />
+      <input
+        type='text'
+        className={clsx('search-bar-input', inputClassName)}
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...rest}
+      />
+    </div>
+  );
+}
+
+type SearchBarResultsProps = React.PropsWithChildren<{ className?: string }>;
+export function SearchBarResults({ children, className }: SearchBarResultsProps) {
+  return <div className={clsx('search-bar-results', className)}>{children}</div>;
+}
```

Comment:
May I ask why this need to be wrap into a component?

Reply:
You can provide it or not. You can style it separately. You can nest it inside something else if you want to. This way just makes it nice and flexible - it's the same pattern that Radix uses a lot.

Reply:
I see, thanks.